frontend/js/src/artist-similarity/src/Graph.tsx

```diff
@@ -0,0 +1,112 @@
+import React, {useState} from 'react';
+import { ResponsiveNetwork, NodeProps, LinkProps, NodeTooltipProps, NetworkSvgProps } from '@nivo/network';
+import { animated, to } from '@react-spring/web';
+import { InputNode } from '@nivo/network';
+interface GraphProps {
+    data: any;
+    fetchData: Function;
+    backgroundColor: string;
+}
+
+const Graph = (props: GraphProps) => {
+    type Node = (typeof props.data)['nodes'][number];
+    type Link = (typeof props.data)['links'][number];
+    const CustomNodeComponent = <Node extends InputNode>({
+        node,
+        animated: animatedProps,
+        onClick,
+        onMouseEnter,
+        onMouseMove,
+        onMouseLeave,
+    }: NodeProps<Node>) => (
+        <animated.g 
+        transform={to([animatedProps.x, animatedProps.y, animatedProps.scale], (x, y, scale) => {
+            return `translate(${x},${y}) scale(${scale})`
+        })}
+        onClick={onClick ? event => onClick(node, event) : undefined}
+        onMouseEnter={onMouseEnter ? event => onMouseEnter(node, event) : undefined}
+        onMouseMove={onMouseMove ? event => onMouseMove(node, event) : undefined}
+        onMouseLeave={onMouseLeave ? event => onMouseLeave(node, event) : undefined}
+        width={to([animatedProps.size], size => size / 2)}
+        height={to([animatedProps.size], size => size / 2)}
+        >
+        <animated.circle
+            data-testid={`node.${node.id}`}
+            r={to([animatedProps.size], size => size / 2)}
+            fill={animatedProps.color}
+            strokeWidth={animatedProps.borderWidth}
+            stroke={animatedProps.borderColor}
+            opacity={animatedProps.opacity}
+        />
+        <animated.text 
+        textAnchor="middle"
+        dominantBaseline="alphabetic"
+        fontSize={to([animatedProps.size], size => size / 6)}
+        style={{ pointerEvents: 'none', fill: 'white'}}
+        >
+            {
+            node.id.split(" ").map((word, index) => (
+                index < 2 ?
+                <animated.tspan x={0} dy={index === 0 ? "0" : to([animatedProps.size], size => size / 6)}>
```

Comment:
Several constants that should be defined here.

Reply:
Fixed.
```
const MAX_LINES = 2;
const MAX_WORD_LENGTH = 10;
```